docs/spec.md

```diff
@@ -4365,8 +4347,11 @@ spaces of `result` and `operand`.
 
 Reverses the order of elements in the `operand` along the specified `dimensions`
 and produces a `result` tensor. More formally,
-`result[i0, ..., ik,..., iR-1] = operand[i0, ..., ik',..., iR-1]` where
-`ik + ik' = dk - 1` for all dimensions `k` in `dimensions`.
+`result[result_index] = operand[operand_index]` where:
```

Comment:
In https://github.com/openxla/stablehlo/pull/1554#discussion_r1210934200, @ghpvnist proposed the following notation: "`result[result_index] = operand[operand_index]` where for `d` in `axes(operand)`", and I'll need to think about it. This will apply to similar ops where we use individual scalars in definition of semantics, so let's discuss this here (since the scope of this PR is larger) instead of the original PR.

Reply:
Agreed that this is worth discussing. However, how about we move this to #289 to facilitate merging this PR, so that quantization PRs can start using it?